src/plugins/embeddable/public/lib/actions/edit_panel_action.ts

```diff
@@ -56,18 +60,35 @@ export class EditPanelAction implements Action<ActionContext> {
 
   public async isCompatible({ embeddable }: ActionContext) {
     const canEditEmbeddable = Boolean(
-      embeddable && embeddable.getOutput().editable && embeddable.getOutput().editUrl
+      embeddable &&
+        embeddable.getOutput().editable &&
+        (embeddable.getOutput().editUrl ||
+          (embeddable.getOutput().editApp && embeddable.getOutput().editPath))
```

Comment:
These `editApp` and `editPath` don't seem to be used now, or are they?

Could we reuse `editUrl` instead of adding above two fields to the embeddable output? We were actually thinking about simplifying embeddable output.

Maybe we could somehow use "url generator" from `share` plugin.

Reply:
If you mean whether any embeddable is currently setting them, you are right. Currently all of them just do hash routing via the `editUrl` output, because they are just used in the same legacy app. The PR cutting over visualize and dashboard to the new platform application service will put them into use.

I thought about leveraging `editUrl` (basically parsing the URL and checking whether they start with `basePath.prepend(/app)`, then plucking out the id and the path and using `navigateToApp` and so on, but I decided to make it explicit because it seemed more stable to me and I didn't want to put so much implicit knowledge about the app URL structure into this action).

Not sure how we could leverage the url generator here, but open for suggestions.

Reply:
> Not sure how we could leverage the url generator here, but open for suggestions.

I'm not sure either, maybe the URL could be created using `share` plugin.

```ts
plugins.share.getUrlGenerator(...).createUrl({ ... });
```

I was just thinking that `.createUrl` method returns a string. So if we used it, we would probably re-use the `.editUrl` field.

Reply:
Created an issue for further discussion on this: https://github.com/elastic/kibana/issues/64497